src/qtpromise/qpromise_p.h

```diff
@@ -87,6 +87,11 @@ static void qtpromise_defer(F&& f)
     qtpromise_defer(std::forward<F>(f), QThread::currentThread());
 }
 
+#ifdef Q_CC_GNU
+#    pragma GCC diagnostic push
+#    pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif
```

Comment:
Making `#pragma` on a whole class looks kinda ugly, but I don't really know how to make it otherwise. One other idea I had was to move `QSharedPointer<T>` initialization from ctor initialization list to a ctor body. That would definitely make it better.